test-lib-openshift.sh

```diff
@@ -344,6 +345,89 @@ function ct_os_cluster_running() {
   oc cluster status &>/dev/null
 }
 
+# ct_os_set_path_oc OC_VERSION
+# --------------------
+# This is a trick that helps using correct version of the `oc`:
+# The input is version of the openshift in format v3.6.0 etc.
+# If the currently available version of oc is not of this version,
+# it first takes a look into /usr/local/oc-<ver>/bin directory,
+# and if not found there it downloads the community release from github.
+# In the end the PATH variable is changed, so the other tests can still use just 'oc'.
+# Arguments: oc_ver - X.Y part of the version of OSE (e.g. 3.9)
+function ct_os_set_path_oc() {
+  local oc_ver=$(ct_os_get_latest_ver $1)
+  local oc_path
+
+  if oc version | grep -q "oc ${oc_ver%.*}." ; then
+    echo "Binary oc found already available in version ${oc_ver}: `which oc` Doing noting."
+    return 0
+  fi
+
+  # first check whether we already have oc available in /usr/local
+  local installed_oc_path="/usr/local/oc-${oc_ver%.*}/bin"
+
+  if [ -x "${installed_oc_path}/oc" ] ; then
+    oc_path="${installed_oc_path}"
+    echo "Binary oc found in ${installed_oc_path}" >&2
+  else
+    # oc not available in /usr/local, try to download it from github (community release)
+    oc_path="/tmp/oc-${oc_ver}-bin"
+    ct_os_download_upstream_oc "${oc_ver}" "${oc_path}"
+  fi
+  if [ -z "${oc_path}/oc" ] ; then
+    echo "ERROR: oc not found installed, nor downloaded" >&1
+    return 1
+  fi
+  export PATH="${oc_path}:${PATH}"
+  if ! oc version | grep -q "oc ${oc_ver%.*}." ; then
+    echo "ERROR: something went wrong, oc located at ${oc_path}, but oc of version ${oc_ver} not found in PATH ($PATH)" >&1
+    return 1
+  else
+    echo "PATH set correctly, binary oc found in version ${oc_ver}: `which oc`"
+  fi
+}
+
+# ct_os_get_latest_ver VERSION_PART_X
+# --------------------
+# Returns full version (vX.Y.Z) from part of the version (X.Y)
+# Arguments: vxy - X.Y part of the version
+# Returns vX.Y.Z variant of the version
+function ct_os_get_latest_ver(){
+  local vxy="v$1"
+  for vz in {3..0} ; do
+    curl -si "https://github.com/openshift/origin/releases/tag/${vxy}.${vz}" | grep -q 'HTTP/1.1 200 OK' && echo "${vxy}.${vz}" && return 0
```

Comment:
`curl -sif ...` can be used instead of `| grep -q ...`.

Reply:
Is `{3..0}` range large enough (for the future)?  It seems like yes looking at the release numbers released so far.

Reply:
So far yes, the largest .z part is 3.7.2, and I wanted to keep it small so we don't do too many requests. Anyway, this way of finding out latest version is really creepy, if anybody would have something that would work similarly, I'm welcoming ideas :)

Reply:
`curl -sif ` looks good, will use it.